File: ui-web/src/lib/game/dailyChallenge.ts
// Daily Challenge Feature

import { getUserInfo } from '../auth.ts';
import { trackEvent } from './analytics';

interface DailyChallenge {
  id: string;
  description: string;
  rewardType: 'cosmetic' | 'currency';
  rewardAmount: number;
  completed: boolean;
}

const dailyChallenges: DailyChallenge[] = [
  {
    id: 'challenge_001',
    description: 'Collect 100 pellets in a single game',
    rewardType: 'currency',
    rewardAmount: 50,
    completed: false
  },
  {
    id: 'challenge_002',
    description: 'Win 3 games as a runner',
    rewardType: 'cosmetic',
    rewardAmount: 1,
    completed: false
  }
];

export async function getDailyChallenge(): Promise<DailyChallenge | null> {
  const user = await getUserInfo();
  if (!user) return null;

  // Fetch or generate daily challenge for the user
  const challenge = dailyChallenges[Math.floor(Math.random() * dailyChallenges.length)];
  return challenge;
}

export function completeChallenge(challengeId: string): void {
  const challenge = dailyChallenges.find(c => c.id === challengeId);
  if (challenge && !challenge.completed) {
    challenge.completed = true;
    trackEvent('challenge_completed', { challengeId });
    // Reward logic here
  }
}

File: core/internal/game/daily_challenge.go
package game

import (
	"time"
)

type DailyChallenge struct {
	ID          string
	Description string
	RewardType  string
	RewardAmount int
	Completed   bool
}

var dailyChallenges = []DailyChallenge{
	{
		ID:          "challenge_001",
		Description: "Collect 100 pellets in a single game",
		RewardType:  "currency",
		RewardAmount: 50,
		Completed:   false,
	},
	{
		ID:          "challenge_002",
		Description: "Win 3 games as a runner",
		RewardType:  "cosmetic",
		RewardAmount: 1,
		Completed:   false,
	},
}

func GetDailyChallenge() DailyChallenge {
	// Randomly select a challenge
	return dailyChallenges[time.Now().Day()%len(dailyChallenges)]
}

func CompleteChallenge(challengeID string) {
	for i, challenge := range dailyChallenges {
		if challenge.ID == challengeID && !challenge.Completed {
			dailyChallenges[i].Completed = true
			// Add reward logic here
		}
	}
}
